addons/box/models.py

```diff
@@ -263,6 +263,7 @@ def after_delete(self, user=None):
         self.deauthorize(Auth(user=user), add_log=True)
         self.save()
 
-    def on_delete(self):
+    def on_delete(self, save=True):
         self.deauthorize(add_log=False)
-        self.save()
+        if save:
+            self.save()
```

Comment:
This was added to the on_delete functions to address John's comment that on_delete would save regardless of the save parameter. There might be a better way to address this though.

Reply:
@corbinSanders @Johnetordoff, but we still need this to save at some point, and you haven't convinced me that this is being saved elsewhere.  

When `save=False` the use case is that we don't want to save an item five times, so we might do save=False in some location, but it is getting saved *later in the process*.   If you're not saving it here, it might not end up getting saved *at all*.  I think you are introducing a bug here.